ASM/c/reset.c

```diff
@@ -0,0 +1,33 @@
+#include "reset.h"
+
+typedef void(*TitleSetup_Init)();
+#define TitleSetup_Init_Func ((TitleSetup_Init) 0x800A0748)
+
+typedef void (*set_bgm)(unsigned int ID);
+#define SET_BGM ((set_bgm) 0x800CAA70)
+
+typedef void (*stop_sfx)(void);
+#define STOP_SFX ((stop_sfx) 0x800A0290)
+
+#define BGM_STOP 0x100000FF
+
+static unsigned char s_reset_delay = 10;
+
+void wait_for_reset_combo(){
+	z64_controller_t pad_pressed = z64_game.common.input[0].raw;
+	if ((pad_pressed.pad.cr) && (pad_pressed.pad.b) && (pad_pressed.pad.s)) {
+		if (s_reset_delay != 0) {
+			s_reset_delay--;
+		}
+		else {
+			STOP_SFX();
+			SET_BGM(BGM_STOP);
+			z64_game.common.state_continue = 0;
+			z64_game.common.next_ctor = (void*)TitleSetup_Init_Func;
+			z64_game.common.next_size = sizeof(z64_game_t);
```

Comment:
The struct used by the "title setup" game state isn't `z64_game_t` (aka global context) so this looks incorrect. I believe this game state doesn't use anything other than the base game state struct, so you can probably just use `z64_ctxt_t`?

Also consider using spaces instead of tabs and maybe using a formatting style that fits the rest of the C code a bit more (tho I guess we lack consistency in general)

Reply:
I guess I don't understand this struct change either. I'd only tested before this commit, but I would be testing a second time after finishing up today's merges.

Reply:
I made the change cause I thought game was gameState and ctx was globalCtx. I will change it back some time today but I tested it as it is and it worked normally.